cdc/scheduler/agent_mock.go

```diff
@@ -0,0 +1,38 @@
+// Copyright 2021 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package scheduler
+
+import (
+	"github.com/pingcap/ticdc/cdc/model"
+	"github.com/pingcap/ticdc/pkg/context"
+)
+
+// ProcessorMessenger implements how messages should be sent to the owner,
+// and should be able to know whether there are any messages not yet acknowledged
+// by the owner.
+type ProcessorMessenger interface {
+	// FinishTableOperation notifies the owner that a table operation has finished.
+	FinishTableOperation(ctx context.Context, tableID model.TableID) (bool, error)
+	// SyncTaskStatuses informs the owner of the processor's current internal state.
+	SyncTaskStatuses(ctx context.Context, running, adding, removing []model.TableID) (bool, error)
+	// SendCheckpoint sends the owner the processor's local watermarks, i.e., checkpoint-ts and resolved-ts.
+	SendCheckpoint(ctx context.Context, checkpointTs model.Ts, resolvedTs model.Ts) (bool, error)
```

Comment:
why need to return a `bool`  here to indicate something, can you explain the potential reason behind it?